src/debugger/nodeDebugWrapper.ts

```diff
@@ -1,194 +1,246 @@
+                                    // and the user needs to Reload JS manually. We prewarm it to prevent that issue
+                                    generator.step("prewarmBundleCache");
+                                    Log.logMessage("Prewarming bundle cache. This may take a while ...");
+                                    return nodeDebugWrapper.remoteExtension.prewarmBundleCache(nodeDebugWrapper.mobilePlatformOptions.platform);
+                                })
+                                .then(() => {
+                                    generator.step("mobilePlatform.runApp");
+                                    Log.logMessage("Building and running application.");
+                                    return mobilePlatform.runApp();
+                                })
+                                .then(() =>
+                                    nodeDebugWrapper.attachRequest(originalRequest, this, request, args, mobilePlatform));
+                        }
+                    }).catch(error =>
+                        nodeDebugWrapper.bailOut(this, error.message));
             });
+        };
+    }
 
-            // We do not permit arbitrary args to be passed to our process
-            args.args = [
-                args.platform,
-                debugServerListeningPort.toString(),
-                !isNullOrUndefined(args.iosRelativeProjectPath) ? args.iosRelativeProjectPath : IOSPlatform.DEFAULT_IOS_PROJECT_RELATIVE_PATH,
-                args.target || "simulator",
-            ];
+    /**
+     * Intecept the "attachRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeAttachRequest(): void {
+        const nodeDebugWrapper = this;
+        this.nodeDebugSession.prototype.attachRequest = function (request: any, args: IAttachRequestArgs) {
+            nodeDebugWrapper.requestSetup(this, args);
+            nodeDebugWrapper.attachRequest(nodeDebugWrapper.nodeDebugSession.prototype.launchRequest, this, request, args, null);
+        };
+    }
 
-            if (!isNullOrUndefined(args.logCatArguments)) { // We add the parameter if it's defined (adapter crashes otherwise)
-                args.args = args.args.concat([parseLogCatArguments(args.logCatArguments)]);
-            }
+    /**
+     * Intecept the "disconnectRequest" instance method of NodeDebugSession to interpret arguments
+     */
+    private customizeDisconnectRequest(): void {
+        const originalRequest = this.nodeDebugSession.prototype.disconnectRequest;
+        const nodeDebugWrapper = this;
+
+        this.nodeDebugSession.prototype.disconnectRequest = function (response: any, args: any): void {
+            // First we tell the extension to stop monitoring the logcat, and then we disconnect the debugging session
+            nodeDebugWrapper.remoteExtension.stopMonitoringLogcat()

```

Comment:
Should this only happen if we started android? What should we do if the following happens:

1. Via the command palette the user runs android, with the implicit logcat output
2. The user debugs against iOS
3. The user stops debugging

Should we stop the logcat monitoring then? I feel like we shouldn't, but this is a niche scenario.